### ISSUE TITLE
Output datasets are not correctly set up when using SDK v2

### ISSUE THREAD

**Issue by ant0nsc** [microsoft/hi-ml#763]
Example job: cyan_mangos_xvqhfz1zqg in hi-ml workspace should have 2 output datasets, but has none.

**Comment by peterhessey** [comment]
It seems the cause of this is issue is in `to_input_dataset_local` in `datasets.py`. The function returns `None, None` when trying to mount v2 datasets as it's apparently not a supported functionality:

```python
        if isinstance(azureml_dataset, FileDataset):
            ...
        else:
            return None, None
```

**Comment by ant0nsc** [comment]
@peterhessey is that resolved by #775 ?

**Comment by peterhessey** [comment]
@ant0nsc I believe so yes, I completed rebuilt the output dataset creation functionality. My above comment is wrong and a different issue I believe.
